### ISSUE TITLE
Oracle connection is not committing data

### ISSUE THREAD

**Issue by 0419** [brettwooldridge/HikariCP#516]
I was not able to find answer in closed issues or on strackoverflow so posting issue here.

I have a oracle connection pooled through HikariCP and it starts to behave strange after a day or two. It executes insert queries fine, and does not fail when commit is called. But no data gets persisted to the database. So I have to re-start the application and do the tasks on fresh connection.

The select queries are working fine and fetching the data, so I know connection is still valid. Just somehow commit is not working, but there is no error.

```
HikariConfig - HikariCP pool HikariPool-0 configuration:
HikariConfig - allowPoolSuspension.............false
HikariConfig - autoCommit......................false
HikariConfig - catalog.........................
HikariConfig - connectionCustomizer............com.zaxxer.hikari.AbstractHikariConfig$1@56cdfb3b
HikariConfig - connectionCustomizerClassName...
HikariConfig - connectionInitSql...............
HikariConfig - connectionTestQuery.............
HikariConfig - connectionTimeout...............30000
HikariConfig - dataSource......................
HikariConfig - dataSourceClassName.............oracle.jdbc.pool.OracleDataSource
HikariConfig - dataSourceJNDI..................
HikariConfig - dataSourceProperties............{password=<masked>, url=jdbc:oracle:thin:@xxx, implicitCachingEnabled=true}
HikariConfig - driverClassName.................
HikariConfig - healthCheckProperties...........{}
HikariConfig - healthCheckRegistry.............
HikariConfig - idleTimeout.....................600000
HikariConfig - initializationFailFast..........true
HikariConfig - isolateInternalQueries..........false
HikariConfig - jdbc4ConnectionTest.............false
HikariConfig - jdbcUrl.........................
HikariConfig - leakDetectionThreshold..........0
HikariConfig - maxLifetime.....................1800000
HikariConfig - maximumPoolSize.................10
HikariConfig - metricRegistry..................
HikariConfig - minimumIdle.....................10
HikariConfig - password........................<masked>
HikariConfig - poolName........................HikariPool-0
HikariConfig - readOnly........................false
HikariConfig - registerMbeans..................false
HikariConfig - threadFactory...................
HikariConfig - transactionIsolation............
HikariConfig - username........................xxx
HikariConfig - validationTimeout...............5000
```

**Comment by brettwooldridge** [comment]
@0419 If that is indeed occurring my guess would be an issue at some layer above HikariCP.  [HikariCP's code](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java#L350) merely calls the underlying ``Connection.commit()``, so it is difficult to see how HikariCP could be the point of failure.

**Comment by 0419** [comment]
yes, I am assuming the same that HikariCP is a proxy to underlying commit. But I wonder if you have any idle check thread in background which is messing up with the connection? 

Another thing to note is I am getting valid update row count after calling ```pstmt.executeUpdate()```, so the query and data are fine, but commit is not working, the idleTimeout setting got me thinking if there is a background thread which is returning the connection to pool again

**Comment by brettwooldridge** [comment]
@0419 HikariCP does not execute any background SQL.  There is no background idle checking.  Connections that are idle for the ``idleTimeout`` are simply closed and removed from the pool.  Connections are tested when they are checked-out from the pool (``getConnection()``).

If you are concerned that these checked might be causing an issue, you can set ``isolateInternalQueries=true``.  But I doubt that is the cause.

If you enable DEBUG logging for the ``com.zaxxer.hikari`` package you might get more information.  For example, if you close a Connection with uncommitted SQL, HikariCP will automatically execute a rollback.  [This will also be logged](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java#L234) at debug level.

**Comment by 0419** [comment]
No significant information was logged, I'll try to switch to another connection pool and see if the issue persists, but cannot test it until next release which is at end of next week. Will let you know

**Comment by 0419** [comment]
Hello Brett, any guess on why following log entry of closing connection might be appearing in log. I am only closing prepared statements and result set objects. The connection object is closed at a later stage in program

```

07:09:16.903 [Hikari Housekeeping Timer (pool HikariPool-0)] DEBUG com.zaxxer.hikari.pool.HikariPool - Before cleanup pool stats HikariPool-0 (total=10, inUse=1, avail=9, waiting=0)
07:09:16.903 [Hikari Housekeeping Timer (pool HikariPool-0)] DEBUG com.zaxxer.hikari.pool.HikariPool - After cleanup pool stats HikariPool-0 (total=9, inUse=1, avail=8, waiting=0)
07:09:16.903 [HikariCP connection closer (pool HikariPool-0)] DEBUG com.zaxxer.hikari.pool.PoolUtilities - Closing connection oracle.jdbc.driver.T4CConnection@6196a075
07:09:16.977 [HikariCP connection filler (pool HikariPool-0)] DEBUG com.zaxxer.hikari.pool.HikariPool - After fill pool stats HikariPool-0 (total=10, inUse=1, avail=9, waiting=0)
```

**Comment by nitincchauhan** [comment]
if you upgrade to 2.4.2, it logs 'reason' for closing connection

**Comment by brettwooldridge** [comment]
@0419 Any update on this issue?

**Comment by 0419** [comment]
No its still closing the connections, I did upgrade the library but its not logging anything specific which can tell reason for closing connection.

Unfortunately I am going to switch to another cp for the Oracle connections to avoid manual restarts of application.

**Comment by nitincchauhan** [comment]
@0419 If you find solution to this issue, please let others here know reason and chosen cp and/or configuration

**Comment by 0419** [comment]
The problem didn't persist after switching to different cp. I am still keeping Hikari for other databases, but for Oracle. Will try the RC1 in few days and keep you posted.

**Comment by nitincchauhan** [comment]
Is there a 'probably commercial' reason not so tell about ''different' cp and its configuration that worked?

**Comment by nitincchauhan** [comment]
good

**Comment by 0419** [comment]
Oh, nothing special, I switched to vibur for oracle

**Comment by 0419** [comment]
should I close this issue? I am unable to allocate time to debug and dont think anyone else is facing it

**Comment by amhamid** [comment]
As an additional information, check if you don't have any caching in between.

**Comment by alecl** [comment]
I saw this in the Oracle docs. Could it have possibly been the reason?

I'm just someone researching for another group so I don't have my own app or Oracle Db to test this myself but figured I'd mention it in case it helps the original poster or others.

https://docs.oracle.com/database/121/JJDBC/drcp.htm#JJDBC29025

> The features of DRCP can be leveraged only with a connection pool on the client because JDBC does not have a default pool on its own. **If you do not have a client connection pool and make any change to the Database with auto commit set to false, then the changes are not committed to the Database while closing the connection.**
